### ISSUE TITLE
Does win10 know about WSL processes?

### ISSUE THREAD

**Issue by heihachi88** [Microsoft/WSL#3455]
I have the case, where do i use proxifier app. I cant track any windows application with it's logs, then add to proxy rules, for example `PING.EXE` - then, when i will use it next time, it'll be using my proxy rule.

But i can't track down processes running inside WSL, for example `ping`, `git` etc. Is there any solution?

**Comment by Voskrese** [comment]
that it is the network or processes?

network by processes in any way, because no daemon is running and the driver for monitoring network traffic from under the kernel is the same

only means windows or the addition of any specific scripts


> to view the processes of many utilities

`gnome-terminal -e htop --title='Системный монитор htop' --maximize`

`sudo gnome-system-monitor`

![2018-08-10_203732](https://user-images.githubusercontent.com/10104740/43958371-c9a80ace-9cdd-11e8-89d4-f0cedb6fccdb.png)

**Comment by Voskrese** [comment]
> interesting project

`https://github.com/wolfc01/procexp`


![2018-08-10_214031](https://user-images.githubusercontent.com/10104740/43961091-9752d780-9ce6-11e8-97f5-2eaa53cd3acf.png)
![2018-08-10_214046](https://user-images.githubusercontent.com/10104740/43961092-97eb6e00-9ce6-11e8-9220-1e00b5768ce1.png)
![2018-08-10_214057](https://user-images.githubusercontent.com/10104740/43961093-97eb6e00-9ce6-11e8-9b70-174d5e249d37.png)
![2018-08-10_214203](https://user-images.githubusercontent.com/10104740/43961094-97eb6e00-9ce6-11e8-800e-959b6498eb38.png)

**Comment by therealkenc** [comment]
> But i can't track down processes running inside WSL, for example ping, git etc. Is there any solution?

Maybe rephrase the question and someone can try to help.

> interesting project

Not interesting enough for a page and a half of screenshots unrelated to WSL.

**Comment by Voskrese** [comment]
@therealkenc  
screenshots are made from windows;) the program is started from under wsl

I as though there is no need to deceive;)

on the design of the window, you can notice that this is VcXsrv

**Comment by heihachi88** [comment]
> Maybe rephrase the question and someone can try to help.

There's an app called Proxifier (socks5 proxy), i want to add rule in it, so my terminal commands, i.e. `git` would be using this proxy rule. But in order to add a program in that rule, you have to find out the process name, if it was just windows process, then i'd just point to that .exe and done. But in WSL there's no exe for `git`.

The question was: how to add a program installed inside WSL (Ubuntu 18.04) to a windows app called Proxifier.

**Comment by 12101111** [comment]
Windows has Win32 subsystem and Windows subsystem of Linux.WSL processes is NT processes but not a Win32 processes. Since Proxifier is a usermode Win32 program,It can't get any Linux side information about a WSL process.You need a usermode Linux program or kernel mode driver like a firewall.

So I suggest you set proxy manually in Linux.

**Comment by heihachi88** [comment]
Could you suggest on how to use socks5 with authentication in linux?

**Comment by 12101111** [comment]
You can try tsocks or proxychains.

**Comment by therealkenc** [comment]
> the program is started from under wsl
> on the design of the window, you can notice that this is VcXsrv

I am aware of that. The point is `procexp` has nothing to do with the Windows Subsystem for Linux, or the ask.

> Could you suggest on how to use socks5 with authentication in linux?

Ah I see. #2340

**Comment by ericblade** [comment]
... is this an accurate description, that WSL operates on a completely different level, without it's connection going through WinSock, and that explains why Proxifier doesn't work at all with WSL processes?  (as an aside, i wonder if that's also the same reason why Edge is completely invisible to Proxifier.. )

I have a separate computer that is setup on my work VPN, which I setup a SSH tunnel to, and use Proxifier to use that singular VPN connection for several different Windows computers.  I haven't yet figured out a way to do the same with live Linux or WSL.

The reason for doing this is twofold: 1- I'm only allowed one VPN connection at a time, but I have multiple computers that I need to be able to connect with, and disconnecting and reconnecting all the time is a serious hassle.  2- it appears that all traffic from the connected machine gets routed through the VPN, or at the very least, all DNS resolution does, which significantly slows down network access. Proxifier provides a method to use local DNS first, and then go to the VPN DNS if the address isn't found, which is much, much faster.

**Comment by hulucc** [comment]
kkcap works with wsl
ji4ozhu/kkcap#3
